app/Http/Controllers/PrioController.php

```diff
@@ -0,0 +1,473 @@
+    }
+
+    /**
+     * Submit priorities for an item
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function submitSingleInput($guildSlug) {
+        $guild         = request()->get('guild');
+        $currentMember = request()->get('currentMember');
+
+        if (!$currentMember->hasPermission('edit.prios')) {
+            request()->session()->flash('status', 'You don\'t have permissions to view that page.');
+            return redirect()->route('member.show', ['guildSlug' => $guild->slug, 'usernameSlug' => $currentMember->slug]);
+        }
+
+        $validationRules =  [
+            'item_id' => 'nullable|integer|exists:items,item_id',
+            'raid_id' => 'required|exists:raids,id',
+            'items.*.characters.id' => 'nullable|integer|exists:characters,id',
+        ];
+
+        $this->validate(request(), $validationRules);
+
+        $guild->load('characters');
+
+        $raid = Raid::where(['guild_id' => $guild->id, 'id' => request()->input('raid_id')])->firstOrFail();
+
+        $itemsWithExistingPrios = Item::
+            where([
+                'items.item_id' => request()->input('item_id'),
+            ])
+            ->with([
+                'priodCharacters' => function ($query) use ($raid) {
+                    return $query->where('character_items.raid_id', $raid->id);
+                },
+            ])
+            ->get();
+
+        $modifiedCount = $this->syncPrios($itemsWithExistingPrios, request()->input('items'), $currentMember, $guild->characters, $raid);
+
+        request()->session()->flash('status', ($modifiedCount ? 'Successfully updated prios for ' : 'No changes made to prios for ') . $itemsWithExistingPrios->first()->name . ' in ' . $raid->name . '.');
+        return redirect()->route('guild.item.show', [
+            'guildSlug' => $guild->slug,
+            'item_id' => $itemsWithExistingPrios->first()->item_id,
+            'slug' => slug($itemsWithExistingPrios->first()->name),
+        ]);
+    }
+
+    /**
+     * A custom sync function that allows for duplicate entries. I didn't see a clear way
+     * to allow duplicates using Laravel's provided sync functions for collections. RIP.
+     *
+     * Heavy on the comments because my brain was having a hard time.
+     *
+     * @param Collection     $itemsWithExistingPrios A collection of items with their current prios
+     * @param Array          $inputItems    The items provided from the HTML form input.
+     * @param App\Member     $currentMember The member syncing these items.
+     * @param App\Characters $characters    The characters we're allowed to attach to.
+     * @param App\Raid       $raid          The raid to associate these prios with.
+     *
+     * @return int The number of items that had their prios modified.
+     */
+    private static function syncPrios($itemsWithExistingPrios, $inputItems, $currentMember, $characters, $raid) {
+        $characters = $characters->keyBy('id');

```

Comment:
Is this creating a hashtable with the key being the id?

Reply:
The table already exists, we're just doing the latter. Replacing the default keys (0, 1, 2, 3) with the ID of each character. Makes referencing the characters a cinch.

Laravel has a pretty powerful object called `Collection` that gets used to store pretty much all arrays of objects. Say I query the database via a model class `Character::all()`; this would get all characters and put them in a `Collection` object containing an array of `Character` objects.

The `Collection` object then allows me to perform about a million extremely useful and handy functions on that collection. https://laravel.com/docs/7.x/collections#available-methods